### ISSUE TITLE
Migrating a jhipster project from 7.9.4 to 8.0.0 using the generator-jhipster-migrate blueprint fails due to @jhipster in generated code

### ISSUE THREAD

**Issue by naris** [jhipster/generator-jhipster-migrate#5]
Migrating a jhipster project from 7.9.4 to 8.0.0 fails with this error due to the presence of @ in code generated by jhipster that is no longer present in the actual code:

`✔ successfully regenerated source application using JHipster 7.9.4
✔ applying prettier to source application
ERROR! An error occured while running jhipster-migrate:migrate#prepareMigrateBranch
ERROR! ERROR! Failed to load plugin 'https://github.com/typescript-eslint' declared in 'CLIOptions': Cannot find module 'typescript'
Require stack:

C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\util\astUtils.js
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\util\index.js
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\rules\adjacent-overload-signatures.js
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\rules\index.js
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\index.js
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@eslint\eslintrc\dist\eslintrc.cjs
Referenced from: CLIOptions
Error: Failed to load plugin 'https://github.com/typescript-eslint' declared in 'CLIOptions': Cannot find module 'typescript'
Require stack:
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\util\astUtils.js
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\util\index.js
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\rules\adjacent-overload-signatures.js
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\rules\index.js
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\index.js
C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@eslint\eslintrc\dist\eslintrc.cjs
Referenced from: CLIOptions
at Module._resolveFilename (node:internal/modules/cjs/loader:1039:15)
at Module._load (node:internal/modules/cjs/loader:885:27)
at Module.require (node:internal/modules/cjs/loader:1105:19)
at require (node:internal/modules/cjs/helpers:103:18)
at Object. (C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules@typescript-eslint\eslint-plugin\dist\util\astUtils.js:30:25)
at Module._compile (node:internal/modules/cjs/loader:1218:14)
at Module._extensions..js (node:internal/modules/cjs/loader:1272:10)
at Module.load (node:internal/modules/cjs/loader:1081:32)
at Module._load (node:internal/modules/cjs/loader:922:12)
at Module.require (node:internal/modules/cjs/loader:1105:19)
WARNING! Error parsing file src\main\webapp\app\home\home.component.html: SyntaxError: Incomplete block "jhipster on Twitter". If you meant to write the @ character, you should use the "@" HTML entity instead. (61:19)
59 | li
60 | a href="https://twitter.com/jhipster" target="_blank" rel="noopener noreferrer" jhiTranslate="home.link.follow"
61 | ->follow https://github.com/jhipster on Twitter<-a
| ^^^^^^^^^^^^^^^^^^^^
62 |
63 | /li
64 | /ul

**Comment by mshima** [comment]
The original issue mention `jhipster-primeng-blueprint` which still in jhipster 7.
Blueprints needs to migrate to jhipster v8.

**Comment by naris** [comment]
@mshima , as far as I know the [jhipster-primeng-blueprint](https://github.com/tailosoft/jhipster-primeng-blueprint) has already been updated to jhipster v8 (in July 2023, before it was released) by the author of the blueprint - [yelhouti](https://github.com/yelhouti)

**Comment by naris** [comment]
@mshima, I generated a new jhipster application, without any plugins using jhipster 7.9.4 and then attempted to migrated it to the latest version of jhipster (8.2.1) using jhipster-migrate and it failed with the same error complaining about @ in home.component.html:
ERROR! An error occured while running jhipster-migrate:migrate#prepareMigrateBranch
ERROR! ERROR! Failed to load plugin '@typescript-eslint' declared in 'CLIOptions': Cannot find module 'typescript'
Require stack:
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\util\astUtils.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\util\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\rules\adjacent-overload-signatures.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\rules\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@eslint\eslintrc\dist\eslintrc.cjs
Referenced from: CLIOptions
Error: Failed to load plugin '@typescript-eslint' declared in 'CLIOptions': Cannot find module 'typescript'
Require stack:
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\util\astUtils.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\util\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\rules\adjacent-overload-signatures.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\rules\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@eslint\eslintrc\dist\eslintrc.cjs
Referenced from: CLIOptions
    at Module._resolveFilename (node:internal/modules/cjs/loader:1039:15)
    at Module._load (node:internal/modules/cjs/loader:885:27)
    at Module.require (node:internal/modules/cjs/loader:1105:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\util\astUtils.js:30:25)
    at Module._compile (node:internal/modules/cjs/loader:1218:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1272:10)
    at Module.load (node:internal/modules/cjs/loader:1081:32)
    at Module._load (node:internal/modules/cjs/loader:922:12)
    at Module.require (node:internal/modules/cjs/loader:1105:19)
WARNING! Error parsing file src\main\webapp\app\home\home.component.html: SyntaxError: Incomplete block "jhipster on Twitter". If you meant to write the @ character, you should use the "&#64;" HTML entity instead. (61:19)
  59 |       li
  60 |         a href="https://twitter.com/jhipster" target="_blank" rel="noopener noreferrer" jhiTranslate="home.link.follow"
> 61 |           follow @jhipster on Twitter/a
     |              ^^^^^^^^^^^^^^^^^^^^
  62 |         
  63 |       /li
  64 |     /ul

**Comment by naris** [comment]
This looks like it is due to this bug in prettier: https://github.com/prettier/prettier/issues/15660

**Comment by naris** [comment]
@mshima The suggestion from the prettier team, and the error message itself is to replace @ with & #64 ; However, since that is in the code newly generated by source version of jhipster, which jihipster-migrate runs, that is not possible. Therefore, how do we work around this bug?

**Comment by mraible** [comment]
@naris I was able to run it on my 21-Points Health project. Currently, the 7.9.3 code is in the [main branch](https://github.com/mraible/21-points). You can view the 7.9.4 upgrade and the 8.2.1 upgrades in the [pull requests](https://github.com/mraible/21-points/pulls).

**Comment by mshima** [comment]
The prettier error should not be fatal.

This error is quite strange:
```
ERROR! ERROR! Failed to load plugin 'https://github.com/typescript-eslint' declared in 'CLIOptions': Cannot find module 'typescript'
```

Added the typescript dependency in https://github.com/jhipster/generator-jhipster/pull/25773

**Comment by mshima** [comment]
You can try reinstalling the migrate plugin or installing locally.

**Comment by naris** [comment]
@mraible is your 21-Points Health project an angular project? (yes it is)
as this seems specific to angular.

@mshima I have tried reinstalling the migrate plugin and installing locally.
I am attempting now to install a local jhipster 7.9.4 and removing the @ jhipster from the default welcome page to see if I can get that to work.

and, no, that did not work either -- it still has the odd typescript CLIOptions error and complaint about @jhipster :(

**Comment by naris** [comment]
@mraible I tried checking out the 21-points health project (main branch) and it also fails to migrate, but for a different reason:
Run `npm audit` for details.
INFO! Using bundled JHipster
node:internal/modules/cjs/loader:544
      throw e;
      ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/util/namespace' is not defined by "exports" in C:\src\21-points\node_modules\yeoman-environment\package.json
    at new NodeError (node:internal/errors:400:5)
    at exportsNotFound (node:internal/modules/esm/resolve:361:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:697:9)
    at resolveExports (node:internal/modules/cjs/loader:538:36)
    at Module._findPath (node:internal/modules/cjs/loader:607:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1025:27)
    at Module._load (node:internal/modules/cjs/loader:885:27)
    at Module.require (node:internal/modules/cjs/loader:1105:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (C:\src\21-points\node_modules\generator-jhipster\utils\blueprint.js:19:25)
    at Module._compile (node:internal/modules/cjs/loader:1218:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1272:10)
    at Module.load (node:internal/modules/cjs/loader:1081:32)
    at Module._load (node:internal/modules/cjs/loader:922:12)
    at Module.require (node:internal/modules/cjs/loader:1105:19)
    at require (node:internal/modules/cjs/helpers:103:18) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}

Node.js v18.13.0
ERROR! successfully regenerated source application using JHipster current
ERROR! An error occured while running jhipster-migrate:migrate#prepareMigrateBranch
ERROR! ERROR! Command failed with exit code 1: npx jhipster --force --skip-install --skip-git --ignore-errors --no-insight --skip-checks --with-entities --prefer-global
Error: Command failed with exit code 1: npx jhipster --force --skip-install --skip-git --ignore-errors --no-insight --skip-checks --with-entities --prefer-global
    at makeError (file:///C:/Users/murwilso/AppData/Roaming/npm/node_modules/generator-jhipster-migrate/node_modules/execa/lib/error.js:60:11)
    at handlePromise (file:///C:/Users/murwilso/AppData/Roaming/npm/node_modules/generator-jhipster-migrate/node_modules/execa/index.js:124:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async default.regenerate (file:///C:/Users/murwilso/AppData/Roaming/npm/node_modules/generator-jhipster-migrate/generators/migrate/generator.js:529:9)
    at async default.prepareMigrateBranch (file:///C:/Users/murwilso/AppData/Roaming/npm/node_modules/generator-jhipster-migrate/generators/migrate/generator.js:322:11)
    at async default.executeTask (file:///C:/Users/murwilso/AppData/Roaming/npm/node_modules/generator-jhipster-migrate/node_modules/yeoman-generator/dist/actions/lifecycle.js:244:13)
    at async runLoop.add.once (file:///C:/Users/murwilso/AppData/Roaming/npm/node_modules/generator-jhipster-migrate/node_modules/yeoman-environment/dist/environment-base.js:395:17) {
  shortMessage: 'Command failed with exit code 1: npx jhipster --force --skip-install --skip-git --ignore-errors --no-insight --skip-checks --with-entities --prefer-global',
  command: 'npx jhipster --force --skip-install --skip-git --ignore-errors --no-insight --skip-checks --with-entities --prefer-global',
  escapedCommand: 'npx jhipster --force --skip-install --skip-git --ignore-errors --no-insight --skip-checks --with-entities --prefer-global',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  cwd: 'C:\\src\\21-points',
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}

**Comment by mraible** [comment]
@naris This is expected since JHipster 7.9.3 requires Node 16. Can you please try with Node 16?

**Comment by naris** [comment]
@mraible I installed node v16.20.2 and now the 21 points health project fails to migrate for the same reason as mine -- @jhipster link in the welcome page:
INFO! Removing C:\src\21-points\src\main\resources\config\tls\keystore.p12
✔ successfully regenerated source application using JHipster current
✔ applying prettier to source application
ERROR! An error occured while running jhipster-migrate:migrate#prepareMigrateBranch
ERROR! ERROR! Failed to load plugin '@typescript-eslint' declared in 'CLIOptions': Cannot find module 'typescript'
Require stack:
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\util\astUtils.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\util\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\rules\adjacent-overload-signatures.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\rules\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@eslint\eslintrc\dist\eslintrc.cjs
Referenced from: CLIOptions
Error: Failed to load plugin '@typescript-eslint' declared in 'CLIOptions': Cannot find module 'typescript'
Require stack:
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\util\astUtils.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\util\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\rules\adjacent-overload-signatures.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\rules\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\index.js
- C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@eslint\eslintrc\dist\eslintrc.cjs
Referenced from: CLIOptions
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1028:15)
    at Function.Module._load (node:internal/modules/cjs/loader:873:27)
    at Module.require (node:internal/modules/cjs/loader:1100:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (C:\Users\murwilso\AppData\Roaming\npm\node_modules\generator-jhipster-migrate\node_modules\@typescript-eslint\eslint-plugin\dist\util\astUtils.js:30:25)
    at Module._compile (node:internal/modules/cjs/loader:1198:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
    at Module.load (node:internal/modules/cjs/loader:1076:32)
    at Function.Module._load (node:internal/modules/cjs/loader:911:12)
    at Module.require (node:internal/modules/cjs/loader:1100:19)
WARNING! Error parsing file src\main\webapp\app\home\home.component.html: SyntaxError: Incomplete block "jhipster on Twitter". If you meant to write the @ character, you should use the "&#64;" HTML entity instead. (66:19)
  64 |       li
  65 |         a href="https://twitter.com/jhipster" target="_blank" rel="noopener noreferrer" jhiTranslate="home.link.follow"
> 66 |           follow @jhipster on Twitter/a
     |                   ^^^^^^^^^^^^^^^^^^^^
  67 |         
  68 |       /li
  69 |     /ul

**Comment by naris** [comment]
@mraible , BTW, building the 21 points health project also fails due to a missing module (chart.js)

./node_modules/ng2-charts/fesm2020/ng2-charts.mjs:3:0-58 - Error: Module not found: Error: Can't resolve 'chart.js' in 'C:\src\21-points\node_modules\ng2-charts\fesm2020'

Error: node_modules/ng2-charts/lib/base-chart.directive.d.ts:2:104 - error TS2307: Cannot find module 'chart.js' or its corresponding type declarations.

2 import { Chart, ChartComponentLike, ChartConfiguration, ChartEvent, ChartType, DefaultDataPoint } from 'chart.js';
                                                                                                         ~~~~~~~~~~


Error: node_modules/ng2-charts/lib/ng-charts.module.d.ts:2:46 - error TS2307: Cannot find module 'chart.js' or its corresponding type declarations.

2 import { ChartComponentLike, Defaults } from 'chart.js';
                                               ~~~~~~~~~~


Error: node_modules/ng2-charts/lib/theme.service.d.ts:2:50 - error TS2307: Cannot find module 'chart.js' or its corresponding type declarations.

2 import { ChartConfiguration, ChartOptions } from 'chart.js';
                                                   ~~~~~~~~~~


Error: src/main/webapp/app/home/home.component.ts:17:50 - error TS2307: Cannot find module 'chart.js' or its corresponding type declarations.

17 import { ChartConfiguration, ChartOptions } from 'chart.js';
                                                    ~~~~~~~~~~




> Task :webapp FAILED

FAILURE: Build failed with an exception.

**Comment by mshima** [comment]
Typescript dependency error should be fixed in v1.0.0

**Comment by mshima** [comment]
I suppose this is fixed.
